src/main/java/com/foalex/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,85 @@
+package com.foalex.bookstore.exception;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    private static final Map<Class<? extends RuntimeException>, HttpStatus> HTTP_STATUS_MAP;
+
+    static {
+        HTTP_STATUS_MAP = new HashMap<>();
+        HTTP_STATUS_MAP.put(EntityNotFoundException.class, HttpStatus.NOT_FOUND);
+        HTTP_STATUS_MAP.put(DataProcessingException.class, HttpStatus.BAD_REQUEST);
+        HTTP_STATUS_MAP.put(DataIntegrityViolationException.class, HttpStatus.CONFLICT);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex, HttpHeaders headers,
+            HttpStatusCode status, WebRequest request
+    ) {
+        Map<String, Object> body = createResponseBody(HttpStatus.BAD_REQUEST,
+                ex.getBindingResult().getAllErrors());
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler(value = EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(RuntimeException ex) {

```

Comment:
this is a handler for `EntityNotFound`

so why the parameter is `RuntimeException`?
```suggestion
    @ExceptionHandler(value = EntityNotFoundException.class)
    protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {

```

Reply:
I have resolved it in Spring Security hw